eclipse/src/saros/editor/internal/EditorAPI.java

```diff
@@ -267,28 +271,274 @@ public static IResource getEditorResource(IEditorPart editorPart) {
+
+    // Skip second part of calculation if no selection is present
+    if (endOffset == offset) {
+      return new TextSelection(startPosition, startPosition);
+    }
+
+    int endLine;
+    int endLineOffset;
+
+    try {
+      /*
+       * Explicitly does not use textSelection.getEndLine() to obtain the start line number as this
+       * call also uses IDocument.getLineOfOffset(...) but simply drops the thrown exception.
+       */
+      endLine = document.getLineOfOffset(endOffset);
+
+      // Skip line offset calculation if end is in same line as start
+      if (endLine == startLine) {
+        endLineOffset = startLineOffset;
+
+      } else {
+        endLineOffset = document.getLineOffset(endLine);
+      }
+
+    } catch (BadLocationException e) {
+      log.warn(
+          "Could not calculate the end line or end line offset for the given end offset "
+              + endOffset
+              + " in the document "
+              + document,
+          e);
+
+      return TextSelection.EMPTY_SELECTION;
+    }
+
+    int endInLineOffset = endOffset - endLineOffset;
+    TextPosition endPosition = new TextPosition(endLine, endInLineOffset);
+
+    return new TextSelection(startPosition, endPosition);
+  }
+
+  /**
+   * Calculates the text based offsets in the given editor part for the given text selection.
+   *
+   * <p>The given editor part must not be <code>null</code> and the given text selection must not be
+   * <code>null</code> and must not be empty.
+   *
+   * @param editorPart the local editor part
+   * @param selection the text selection
+   * @return the absolute offsets in the given editor part for the given text selection or {@link
+   *     org.eclipse.jface.text.TextSelection#emptySelection()} if the given editor is not a text
+   *     editor, the editor does not have a valid document provider, a valid IDocument could not be
+   *     obtained form the document provider, or the line offsets could not be calculated
+   * @throws NullPointerException if the given editor part or selection is <code>null</code>
+   * @throws IllegalArgumentException if the given text selection is empty
+   */
+  public static ITextSelection calculateOffsets(IEditorPart editorPart, TextSelection selection) {
+
+    Objects.requireNonNull(editorPart, "The given editor part must not be null");
+    Objects.requireNonNull(selection, "The given selection must not be null");
+
+    if (selection.isEmpty()) {
+      throw new IllegalArgumentException("The given text selection must not be empty");
+    }
+
+    if (!(editorPart instanceof ITextEditor)) {
+      return org.eclipse.jface.text.TextSelection.emptySelection();
+    }
+
+    ITextEditor textEditor = (ITextEditor) editorPart;
+
+    IDocumentProvider documentProvider = textEditor.getDocumentProvider();
+
+    if (documentProvider == null) {
+      return org.eclipse.jface.text.TextSelection.emptySelection();
+    }
```

Comment:
I think that this is heavily subjective. I am not consistent on the handling either.

For me, this is a separate path of the code flow (the early exit for no valid document provider) and therefore is in its own block. But I have to admit that I sometimes overdo it with the blank lines in my code.

I still don't think that it is necessary to change this here, especially because this format matches the styling of (most of) the rest of the file.